packages/reports/addon/models/bard-request/request.js

```diff
@@ -459,6 +460,28 @@ export default Fragment.extend(Validations, {
     return this.removeSort(sortObj);
   },
 
+  /**
+   * Removes the sort from the sort array using the metric model and parameters
+   *
+   * @method removeSortMetricWithParam
+   * @param {Object} metricModel - metric metadata model
+   * @param {Object} [parameters] - parameters [optional]
+   * @returns {Void}
+   */
+  removeSortMetricWithParam(metricModel, parameters) {
+    let sorts = get(this, 'sort').filterBy('metric.metric', metricModel),
+        canonicalizedMetric = canonicalizeMetric({
+          metric: get(metricModel, 'name'),
+          parameters
+        });
+
+    sorts.forEach(sortMetric => {
+      if(!parameters || get(sortMetric, 'metric.canonicalName') === canonicalizedMetric) {
```

Comment:
I still feel that removing a metric without parameters should not be a part of this method.

https://gist.github.com/balajimanoharan/9f6319eb0371cf72ad4099294219fba3
Would something like this work?

Reply:
Yeah looks good, I'll refactor